packages/frontend-api/src/Model/Parameter/ParameterWithValuesFactory.php

```diff
@@ -63,4 +63,27 @@ public function createMultipleForProduct(Product $product): array
 
         return $parametersWithValues;
     }
+
+    /**
+     * @param array $productData
+     * @return array
+     */
+    public function createParameterArrayFromProductArray(array $productData): array
+    {
+        $parametersWithValues = [];
+
+        foreach ($productData['parameters'] as $product) {
+            $parametersWithValues[] = [
```

Comment:
From previous comment: Please consider extracting to `createParameterArrayFromProductArray`.

Reply:
okay, in 9be1288de